### PR TITLE
feat: add harper-ls

### PR DESCRIPTION
**Pull request by elijah-potter**
Hey. I've been working on an English grammar checker called Harper.
I've recently gotten approval to add `harper-ls` (the LSP frontend) to `nvim-lspconfig`. However, I've found that many users find lack of Mason support a significant deterrent, so I'd like to add it to the registry.

I wrote the `package.yaml` in reference to `rust-analyzer/package.yaml`, and I believe it is spec-compliant.

You can find out more about it [here](https://github.com/chilipepperhott/harper).

Thank you for your consideration.

### CODE REVIEW

**Comment by chilipepperhott** [comment]
The main difference between this pull request and #4590 is that this one uses Harper's official release binaries.

File: packages/harper-ls/package.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+name: harper-ls
+description: The Grammar Checker for Developers
+homepage: https://github.com/chilipepperhott/harper-ls
+licenses:
+  - Apache-2.0
+languages:
+  - Markdown,
```
Comment (chrisgrieser) [inline packages/harper-ls/package.yaml:8]: just a heads-up: the trailing commas are unnecessary, in fact they just make the display in the mason window a bit uglier:
Pasted image 2024-03-05 at 11 10 25@2x

They don't make the package invalid though, just an visual issue

File: packages/harper-ls/package.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  - Rust,
+  - TypeScript,
+  - JavasCript,
```
Comment (chrisgrieser) [inline packages/harper-ls/package.yaml:11]: typo (wrong capitalization)

File: packages/harper-ls/package.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  - Python,
+  - Go,
+  - C,
+  - C++,
+  - Ruby,
+  - Swift,
```
Comment (chrisgrieser) [inline packages/harper-ls/package.yaml:17]: Swift is not an accepted filetype (will make the check fail, when it is triggered).

If you install the yaml lsp, you will get completion and diagnostics that validate for the mason packages (which is how I could tell)

File: packages/harper-ls/package.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
+---
+name: harper-ls
+description: The Grammar Checker for Developers
+homepage: https://github.com/chilipepperhott/harper-ls
```
Comment (chrisgrieser) [inline packages/harper-ls/package.yaml:4]: this URL does not exist anymore

File: packages/harper-ls/package.yaml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  - Swift,
+  - C#,
+  - TOML,
+  - Lua,
+categories:
+  - LSP
+
+source:
+  # renovate:versioning=loose
+  id: pkg:github/chilipepperhott/harper-ls@v0.7.3
```
Comment (chrisgrieser) [inline packages/harper-ls/package.yaml:26]: due to the change of the repo location, this is invalid and will result on the download failing

**Comment by chrisgrieser** [comment]
Just added the yaml file to my personal registry. There are a bunch of issues, some of which make the package invalid.

**Comment by elijah-potter** [comment]
Thank you so much for taking a look.
I've gone ahead and fixed the issues you've highlighted. I'm really sorry about that. 
I'm honestly quite embarrassed.

**Comment by chrisgrieser** [comment]
All good, my first PR to the registry also has a bunch of issues. 

Package should be good to merge and pass the checks now 👍

**Review by williamboman — APPROVED** [review]
Thanks! I updated the Windows asset file as there doesn't seem to be a `-msvc.zip` variant. The `loose` versioning override is also not needed because they use semantic versioning (default). Also thanks @chrisgrieser for helping reviewing

**Pull request merged**